### PR TITLE
Bumped react & react-dom to 16.8.6

### PR DESCRIPTION
**Pull request by tonyanziano**
This is in preparation for when we bump to the next version of Web Chat.

- bumped `react` & `react-dom` to `^16.8.6` to sync with Web Chat
- fixed bugs that surfaced due to the React bump (`OpenBotDialog.tsx`)
- fixed a bug in the `<AutoComplete />` component (also tied to `OpenBotDialog.tsx`)

### CODE REVIEW

File: packages/app/client/src/vendors.ts [hunk 1]
```diff
@@ -30,12 +30,12 @@
...
 //
+
 import 'base64url';
 import 'botframework-config/lib/models';
 import 'botframework-config/lib/schema';
 import 'botframework-webchat';
 import 'react';
 import 'react-dom';
 import 'redux';
-import 'redux-promise-middleware';
```
Comment (tonyanziano) [inline packages/app/client/src/vendors.ts]: this was removed as a dependency in #1721 ([line where it was removed](https://github.com/microsoft/BotFramework-Emulator/pull/1721/files#diff-0eb7daa45f59d0300af0e962e588905bL120))

**Comment by tonyanziano** [comment]
A couple of tests blew up. Going to fix those.

**Comment by tonyanziano** [comment]
Need to fix a few components that were relying on (unknowingly) broken `getDerivedStateFromProps()` functions

File: packages/app/client/src/ui/shell/explorer/resourceExplorer/resourceExplorer.tsx [hunk 2]
```diff
@@ -51,21 +51,29 @@ function simpleNameSort(a: IFileService, b: IFileService): 0 | 1 | -1 {
...
   openResource: (resource: IFileService) => void;
   resourcesPath?: string;
   openResourcesSettings?: (dialog: ComponentClass<any>) => void;
 }
 
 export class ResourceExplorer extends ServicePane<ResourceExplorerProps, ResourceExplorerState> {
-  public static getDerivedStateFromProps(newProps: ResourceExplorerProps) {
-    const { fileToRename = {} } = newProps;
-    return { fileToRename: { ...fileToRename } }; // Copies only
+  // public static getDerivedStateFromProps(newProps: ResourceExplorerProps) {
```
Comment (cwhitten) [inline packages/app/client/src/ui/shell/explorer/resourceExplorer/resourceExplorer.tsx]: remove?

Comment (tonyanziano) [reply packages/app/client/src/ui/shell/explorer/resourceExplorer/resourceExplorer.tsx]: good catch

**Pull request merged**